syntax = "proto3";

package ai_service;

service LLMAgentSubService {
    rpc InvokeInterpreterAgent(AIServiceTaskRequest) returns (AIServiceAcknowledgement) {}
    rpc InvokeRetrievalAgent(AIServiceTaskRequest) returns (AIServiceAcknowledgement) {}
    rpc InvokeSummarizationAgent(AIServiceTaskRequest) returns (AIServiceAcknowledgement) {}
    rpc InvokeResponseAgent(AIServiceTaskRequest) returns (AIServiceAcknowledgement) {}
}


message AIServiceTaskRequest {
    string task_id = 1;
    string job_id = 2;
}

message AIServiceAcknowledgement {
    bool success = 1;
}